### ISSUE TITLE
[Bug]: Samba server won't honnor max log file = 50

### ISSUE THREAD

**Issue by bugith** [termux/termux-packages#21899]
### Problem description

This morning I find my phone is out of storage.
du -a .| sort -nr| head -n20 ran from com.termux/files dir shows the culprit is usr/var/log/samba/log.smbd (190GB)

Removed then restarting some crashed apps and everything seems OK.

### What steps will reproduce the bug?

I imagine maybe just restart smbd -i and let run until phone turns into sleep mode or something similar, because I saw twice the termux screen spitting error messages I couldn't stop (Ctrl+C did not work) and I had to force-kill from Android Settings Application

### What is the expected behavior?

Don't fill the storage, respect the directive

### System information

```shell
Termux Variables:
TERMUX_APK_RELEASE=F_DROID
TERMUX_APP_PACKAGE_MANAGER=apt
TERMUX_APP_PID=15607
TERMUX_IS_DEBUGGABLE_BUILD=0
TERMUX_MAIN_PACKAGE_FORMAT=debian
TERMUX_VERSION=0.118.1
TERMUX__USER_ID=0
Packages CPU architecture:
aarch64
Subscribed repositories:
# sources.list
deb https://packages-cf.termux.dev/apt/termux-main/ stable main
Updatable packages:
c-ares/stable 1.34.2 aarch64 [upgradable from: 1.34.1]
command-not-found/stable 2.4.0-47 aarch64 [upgradable from: 2.4.0-45]
libnettle/stable 3.10 aarch64 [upgradable from: 3.9.1]
libssh2/stable 1.11.1 aarch64 [upgradable from: 1.11.0]
libtirpc/stable 1.3.6 aarch64 [upgradable from: 1.3.5]
libunbound/stable 1.22.0 aarch64 [upgradable from: 1.21.1-1]
libunistring/stable 1.3 aarch64 [upgradable from: 1.1]
termux-am/stable 0.8.0-1 all [upgradable from: 0.8.0]
termux-tools/stable 1.44.1 aarch64 [upgradable from: 1.43.6]
termux-tools version:
1.43.6
Android version:
14
Kernel build information:
Linux localhost 5.4.242-qgki-g3f7ff67280a0 #1 SMP PREEMPT Fri Jul 5 16:49:09 CST 2024 aarch64 Android
Device manufacturer:
Nothing
Device model:
A063
LD Variables:
LD_LIBRARY_PATH=
LD_PRELOAD=/data/data/com.termux/files/usr/lib/libtermux-exec.so
Installed termux plugins:
com.termux.widget versionCode:13
```

**Comment by bugith** [comment]
Some more info : this happened too with smbd -D
testparm output, followed with log file head:
# Global parameters
[global]
	add group script = /data/data/com.termux/files/usr/bin/true
	add machine script = /data/data/com.termux/files/usr/bin/true
	add user script = /data/data/com.termux/files/usr/bin/true
	add user to group script = /data/data/com.termux/files/usr/bin/true
	allow dns updates = disabled
	bind interfaces only = Yes
	dcerpc endpoint servers = rpcecho
	deadtime = 15
	delete group script = /data/data/com.termux/files/usr/bin/true
	delete user from group script = /data/data/com.termux/files/usr/bin/true
	delete user script = /data/data/com.termux/files/usr/bin/true
	disable spoolss = Yes
	dns proxy = No
	dns update command = /data/data/com.termux/files/usr/bin/false
	enable core files = No
	host msdfs = No
	interfaces = 127.0.0.0/8 wlan0
	lm announce = No
	load printers = No
	local master = No
	max log size = 50
	max smbd processes = 8
	max stat cache size = 1024
	min receivefile size = 16384
	multicast dns register = No
	name resolve order = lmhosts host bcast
	passwd program = /data/data/com.termux/files/usr/bin/true
	preferred master = No
	printcap name = /dev/null
	rename user script = /data/data/com.termux/files/usr/bin/true
	restrict anonymous = 2
	security = USER
	server role = standalone server
	server services = smb
	server string = Samba on Termux
	show add printer wizard = No
	smb ports = 1139 4445
	socket options = TCP_NODELAY SO_KEEPALIVE
	unix extensions = No
	idmap config * : backend = tdb
	aio write size = 0
	delete readonly = Yes
	ea support = No
	inherit permissions = Yes
	lpq command = /data/data/com.termux/files/usr/bin/false
	lprm command = /data/data/com.termux/files/usr/bin/false
	map archive = No
	max connections = 6
	max print jobs = 0
	nt acl support = No
	print command = /data/data/com.termux/files/usr/bin/false
	store dos attributes = No
	strict locking = No
	use sendfile = Yes


[internal]
	browseable = No
	comment = Internal storage
	path = @TERMUX_HOME@/storage/shared
	read only = No
	vfs objects = aio_pthread
	aio_pthread:aio open = yes

The log file is flooded with these messages:
[2024/10/20 11:44:39.459452,  0] ../../source3/smbd/server.c:1741(main)
  smbd version 4.16.11 started.
  Copyright Andrew Tridgell and the Samba Team 1992-2022
[2024/10/20 11:54:50.435054,  0] ../../source3/smbd/server.c:981(smbd_accept_connection)
  accept: Invalid argument
[2024/10/20 11:54:50.435805,  0] ../../source3/smbd/server.c:981(smbd_accept_connection)
  accept: Invalid argument
[2024/10/20 11:54:50.435994,  0] ../../source3/smbd/server.c:981(smbd_accept_connection)
  accept: Invalid argument
[2024/10/20 11:54:50.436238,  0] ../../source3/smbd/server.c:981(smbd_accept_connection)
  accept: Invalid argument
[2024/10/20 11:54:50.436428,  0] ../../source3/smbd/server.c:981(smbd_accept_connection)
  accept: Invalid argument
[2024/10/20 11:54:50.436722,  0] ../../source3/smbd/server.c:981(smbd_accept_connection)
  accept: Invalid argument
[2024/10/20 11:54:50.436898,  0] ../../source3/smbd/server.c:981(smbd_accept_connection)
  accept: Invalid argument
[2024/10/20 11:54:50.437073,  0] ../../source3/smbd/server.c:981(smbd_accept_connection)
  accept: Invalid argument
[2024/10/20 11:54:50.437245,  0] ../../source3/smbd/server.c:981(smbd_accept_connection)
...

**Comment by pizhd** [comment]
I also have the same problem，the log related configuration seems to be ineffective

**Comment by truboxl** [comment]
Please help download, install the artifact and check https://github.com/termux/termux-packages/actions/runs/12093861029?pr=22425 if it is resolved

**Comment by truboxl** [comment]
Made a mistake in the patch. Please test the update artifact https://github.com/termux/termux-packages/actions/runs/12150475554.

**Comment by bugith** [comment]
> Please help download, install the artifact and check https://github.com/termux/termux-packages/actions/runs/12093861029?pr=22425 if it is resolved

Hi @truboxl
I upgraded with
`pkg upgrade samba`
The issue is the same. When I run the upgrade again after your patch published below I get a message stating _samba is already the newest version (4.16.11-2)._
`ls -l log.smbd
-rw-r--r--. 1 u0_a299 u0_a299 144940141439 Dec  8 19:26 usr/var/log/samba/log.smbd`

`smbd -V
Version 4.16.11`
Trying to running smbd as root I get
`[2024/12/06 18:40:06,  0] ../../source3/lib/dumpcore.c:60(get_default_corepath)
  Failed to create /data/data/com.termux/files/usr/var/log/samba/cores for user 0 with mode 0700
[2024/12/06 18:40:06,  0] ../../source3/lib/dumpcore.c:257(dump_core_setup)
  Unable to setup corepath for smbd: File exists`

**Comment by bugith** [comment]
Hmmm, I found the arm zip with checksum and relative path to 4.16.11-3 but I'm afraid I don't know how to install this file.

**Comment by bugith** [comment]
Patch is OK, it seems to work: I downsized the log size directive to 20 (which stands for kB), and after some smbd restarts log.smbd grew just above 20k (20515) and was then moved to log.smbd.old, and a new log.smbd was created. Lets check what happens when the new log.smbd reaches the threshold...
Although, on first time I restarted (and only this time) I noticed, running many `ls -l path/log.smbd` in one minute, that the log file was growing up and down as if we had some race condition. I believe I saw this size going above the directive, but not sure. I noticed this behaviour just after I went back to user terminal after a short switch to `su` whose aim was to check if the log was still flooded with `../../source3/smbd/server.c:981(smbd_accept_connection)` messages as stated above, what I think know looks like a DoS attack.
